entitlement/src/main/java/org/killbill/billing/entitlement/dao/DefaultBlockingStateDao.java

```diff
@@ -258,17 +262,21 @@ public Void inTransaction(final EntitySqlDaoWrapperFactory entitySqlDaoWrapperFa
 
                     final BlockingAggregator currentState = getBlockedStatus(sqlDao, entitySqlDaoWrapperFactory.getHandle(), state.getBlockedId(), state.getType(), bundleId, upToDate, context);
                     if (previousState != null && currentState != null) {
-                        recordBusOrFutureNotificationFromTransaction(entitySqlDaoWrapperFactory,
-                                                                     state.getId(),
-                                                                     state.getEffectiveDate(),
-                                                                     state.getBlockedId(),
-                                                                     state.getType(),
-                                                                     state.getStateName(),
-                                                                     state.getService(),
-                                                                     inserted,
-                                                                     previousState,
-                                                                     currentState,
-                                                                     context);
+                        final boolean isBusEvent = state.getEffectiveDate().compareTo(context.getCreatedDate()) <= 0;
```

Comment:
Nit: I would have kept the `isBusEvent` logic encapsulated in `recordBusOrFutureNotificationFromTransaction` and passed the `seqId` by reference (or simply pass the value and update the signature to return the new value).

Reply:
I wanted to keep the logic very similar to what we have in the [subscription layer](https://github.com/sbrossie/killbill/blob/events-create-subscriptions-optimization/subscription/src/main/java/org/killbill/billing/subscription/engine/dao/DefaultSubscriptionDao.java#L611), and while here we could abstract this inside the `recordBusOrFutureNotificationFromTransaction` as you suggested, it would not work (easily) in the subscription layer as this method is shared. Happy to revisit...